### PR TITLE
Add support for currentDirectoryPath in Process class

### PR DESCRIPTION
**Pull request by mattrajca**
This patch adds support for the `currentDirectoryPath` property on `Process`

Since `posix_spawn` doesn't make it easy to change the working directory, a
call to `chdir` is issued just before the process executes. It looks like
`lldb` follows a similar pattern in one of its tools
(`tools/darwin-debug/darwin-debug.cpp`).

This also addresses [SR-3524](https://bugs.swift.org/browse/SR-3524)

### CODE REVIEW

File: Foundation/Process.swift [hunk 1]
```diff
@@ -387,6 +387,30 @@ open class Process: NSObject {
...
         }
 
+        let currentDirectoryURL = URL(fileURLWithPath: currentDirectoryPath)
+        let changeDirectorySucceeded = currentDirectoryURL.withUnsafeFileSystemRepresentation { rawPath -> Bool in
+            guard let rawPath = rawPath else {
+                return false
+            }
+
+            // posix_spawn doesn't support passing a custom working directory, so resort to chdir.
+            return chdir(rawPath) == 0
```
Comment (parkera) [inline Foundation/Process.swift]: Am I missing something, or don't we have to reset this value after we're done with the launch?

Comment (mattrajca) [inline Foundation/Process.swift]: @parkera Good catch, I updated the commit.

**Comment by mattrajca** [comment]
@parkera please review the revised commit.

**Comment by parkera** [comment]
Looks good, thanks. Can you just do one more thing for me and add a couple lines to the test to 1. get the CWD before/after and comparing, to verify that we did that correctly too.

**Comment by mattrajca** [comment]
@parkera: Done! Please re-review the latest commit.

**Comment by alblue** [comment]
@swift-ci please test

**Comment by mattrajca** [comment]
@parkera Good to go?

**Comment by parkera** [comment]
Yes, thanks for taking this one on!

**Comment by parkera** [comment]
@swift-ci test and merge

**Pull request merged**